File: benchmarks/xamarin/Droid/GraphicsActivity.cs

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Java.Interop;
using Java.Lang;
using Android.Graphics;

namespace xamarin.Droid
{
    [Activity(Label = "GraphicsActivity")]			
    public class GraphicsActivity : Activity
    {
        private TextView benchmarkResult;
        private Button runBtn;
        private SurfaceView surfaceView;

        protected override void OnCreate(Bundle bundle)
        {
            base.OnCreate(bundle);
            SetContentView(Resource.Layout.activity_graphics);

            benchmarkResult = FindViewById<TextView>(Resource.Id.benchmark_result_tv);
            runBtn = FindViewById<Button>(Resource.Id.run_btn);
            surfaceView = FindViewById<SurfaceView>(Resource.Id.surfaceView);
        }

        [Export("run")]
        public void run(View view) {          
            runBtn.Enabled = false;
            TrialSuite ts = new TrialSuite(1500);
            for (int times = 0; times < 1500; times++) {
                Canvas canvas = surfaceView.Holder.LockCanvas();

                Paint paint = new Paint(PaintFlags.AntiAlias);
                paint.Color = new Color(times % 255, 255 - times % 255, 0);
                paint.StrokeWidth = 1;

                long now = JavaSystem.NanoTime();

                long a = testableMethod(canvas, paint);

                long end = JavaSystem.NanoTime();

                surfaceView.Holder.UnlockCanvasAndPost(canvas);

                ts.addTrial(now, end);
            }
            Console.WriteLine(ts.getComaSeparatedData());
            string str = ts.getAverage() + "\n";                   
            benchmarkResult.Append(str);
            runBtn.Enabled = true;                    
        }

        private long testableMethod(Canvas canvas1, Paint paint) {
            int w = canvas1.Width;
            int h = canvas1.Height;
            for (int x = 0; x < w; x++) {
                for (int y = 0; y < h; y++) {
                    canvas1.DrawPoint(x, y, paint);
                }
            }
            return w;
        }
    }
}

File: benchmarks/custom/client-android/src/main/java/com/github/larchaon/android/GraphicsActivity.java
package com.github.larchaon.android;

import android.app.Activity;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.os.Bundle;
import android.view.SurfaceView;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import com.github.larchaon.shared.TrialSuite;

public class GraphicsActivity extends Activity {

    private TextView benchmarkResult;
    private Button runBtn;
    private SurfaceView surfaceView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_graphics);

        benchmarkResult = (TextView) findViewById(R.id.benchmark_result_tv);
        runBtn = ((Button) findViewById(R.id.run_btn));
        surfaceView = ((SurfaceView) findViewById(R.id.surfaceView));
    }

    public void run(View view) {
        new Thread(new Runnable() {
            @Override
            public void run() {
                runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        runBtn.setEnabled(false);
                    }
                });

                TrialSuite ts = new TrialSuite(1500);
                for (int times = 0; times < 1500; times++) {
                    Canvas canvas = GraphicsActivity.this.surfaceView.getHolder().lockCanvas();

                    Paint paint = new Paint(Paint.ANTI_ALIAS_FLAG);
                    paint.setColor(0xff000000 + (times % 256 << 16) + (256 - times % 256 << 8));
                    paint.setStyle(Paint.Style.STROKE);
                    paint.setStrokeWidth(1);

                    long now = System.nanoTime();

                    long a = testableMethod(canvas, paint);

                    long end = System.nanoTime();

                    surfaceView.getHolder().unlockCanvasAndPost(canvas);

                    ts.addTrial(now, end);
                }
                final String str = ts.getAverage() + "\n";
                System.out.println(ts.getComaSeparatedData());
                runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        benchmarkResult.append(str);
                        runBtn.setEnabled(true);
                    }
                });
            }
        }).start();
    }

    private long testableMethod(Canvas canvas1, Paint paint) {
        int w = canvas1.getWidth();
        int h = canvas1.getHeight();
        for (int x = 0; x < w; x++) {
            for (int y = 0; y < h; y++) {
                canvas1.drawPoint(x, y, paint);
            }
        }
        return w;
    }
}
